NewForce-Cohort-8/tabloidfullstack-suplex-syndicate
# View all posts component, ticket 3

**Description:**
Trello ticket 3: https://trello.com/c/Qj1NskH3/3-3-view-all-posts

This ticket allows the logged in user to click on the "Posts" link in the navbar and be directed to a list of all the posts that have been approved, sorted with the most recent on top. Each post also includes the title, author, and category Id (after the category component has been pushed up, I will change the category Id on each post to the actual category name).

**Tasks done for this ticket:**
-  Post Model, Post Repository, IPost Repository, and Post Controller were built on the server-side to accomplish a successful API request to get all posts.
- Post.js, PostList.js, and PostManager.js were built out on the client-side.
- "/post" route path added to Application Views as well as linked in the navbar on the client-side. (I will update this past to be "/posts" instead of "/post" on the next pull request.

**Type of change**

**Testing Instructions**
1. git add and git commit whatever you're working on on your branch.
2. git checkout main.
3. git fetch -a.
4. git checkout my branch: cb-beginning.
5. If you don't see the most recent version of my files, git pull origin cb-beginning.
6. (if not already done) Please create the Tabloid-Fullstack **database** by running the SQL script in a SQL query in Visual Studio 2022. The script is located here: https://github.com/NewForce-Cohort-8/tabloidfullstack-suplex-syndicate/blob/main/SQL/01_Tabloid_Create_DB.sql. 
7. (if not already done) Add the **seed data** for the project by running the SQL script in a SQL query in Visual Studio 2022. The script is located here: https://github.com/NewForce-Cohort-8/tabloidfullstack-suplex-syndicate/blob/main/SQL/02_Tabloid_Seed_Data.sql.
8. Please right click on the new Tabloid database in the SQL Server Object Explorer in Visual Studio 2022 (you may need to refresh after adding the data) and select "New Query" to **open a new query**.
9. (if not already done) **Install your dependencies** by running npm install from the same directory as your package.json file
10. Please run the following statement in the query to get a list of tags in the database: **"SELECT * FROM Post WHERE IsApproved = 1 AND PublishDateTime < SYSDATETIME() ORDER BY PublishDateTime DESC;"**. This is the list of posts you'll be working with.
11. After you know what posts are in the DB, **open the TabloidFullStack.sln project** in Visual Studio 2022 (if not already open) and **launch the API** by clicking the execute with debugger solid green button.
12. Open a new GitBash window and CD into the client folder of the project and **open Visual Studio Code** with the command "code .", then launch the react app with the command "npm start". This should open in your browser.
13. In the react app, if you are not already logged in, you should be directed to the login page. Please **log in** with this email address: "foo@bar.com". This should redirect you back to the home page.
14. On nav bar, **click "Posts"**. You should be directed to the list of posts described at the top of this pull request. **Please verify that these match the list in your query**.

Checklist:
